File: src/main/java/org/minturtle/careersupport/user/dto/UserInfoResponse.java
package org.minturtle.careersupport.user.dto;


import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.minturtle.careersupport.user.entity.User;

@AllArgsConstructor
@Getter
@NoArgsConstructor
public class UserInfoResponse {

    private String nickname;

    public static UserInfoResponse of(User user){
        return new UserInfoResponse(user.getNickname());
    }

    public static UserInfoResponse of(UserInfoDto user){
        return new UserInfoResponse(user.getNickname());
    }
}

File: src/main/java/org/minturtle/careersupport/user/dto/UserInfoResponse.kt
package org.minturtle.careersupport.user.dto

import org.minturtle.careersupport.user.entity.User

data class UserInfoResponse(
    val nickname: String
){

    companion object{
        @JvmStatic
        fun of(user: User): UserInfoResponse {
            return UserInfoResponse(user.nickname)
        }

        @JvmStatic
        fun of(user: UserInfoDto): UserInfoResponse {
            return UserInfoResponse(user.nickname)
        }
    }

}